### PR TITLE
removed sparse euclidean distance as no longer required

### PR DESCRIPTION
**Pull request by lambday**
- `CSparseFeatures` instances now work with plain `CEuclideanDistance`. This would enable us using sparse features in GaussianKernel, for example.

### CODE REVIEW

**Comment by lambday** [comment]
@karlnapf @sonney2k @lisitsyn @vigsterkr please comment!

**Comment by karlnapf** [comment]
That is, since EuclideanDistance can deal with sparse vectors, this is not needed anymore?

**Comment by lambday** [comment]
@karlnapf exactly! Realized that our `CSparseFeatures` is an implementation of `CDotFeatures`.

File: src/shogun/distance/SparseEuclideanDistance.cpp [hunk 1]
```diff
@@ -1,81 +0,0 @@
...
-}
-
-bool CSparseEuclideanDistance::init(CFeatures* l, CFeatures* r)
-{
-	CSparseDistance<float64_t>::init(l, r);
-
-	cleanup();
-
-	sq_lhs=SG_MALLOC(float64_t, lhs->get_num_vectors());
-	sq_lhs=((CSparseFeatures<float64_t>*) lhs)->compute_squared(sq_lhs);
```
Comment (karlnapf) [inline src/shogun/distance/SparseEuclideanDistance.cpp]: this kind of stuff is also not slower/memory intense now?

File: src/shogun/distance/SparseEuclideanDistance.cpp [hunk 2]
```diff
@@ -1,81 +0,0 @@
...
-		SG_FREE(sq_rhs);
-	sq_rhs = NULL;
-
-	SG_FREE(sq_lhs);
-	sq_lhs = NULL;
-}
-
-float64_t CSparseEuclideanDistance::compute(int32_t idx_a, int32_t idx_b)
-{
-	float64_t result=((CSparseFeatures<float64_t>*) lhs)->compute_squared_norm(
```
Comment (karlnapf) [inline src/shogun/distance/SparseEuclideanDistance.cpp]: and this is also still fine?

File: src/shogun/ui/SGInterface.cpp [hunk 3]
```diff
@@ -4293,12 +4293,9 @@ bool CSGInterface::cmd_set_distance()
 	{
 		distance=ui_distance->create_generic(D_BRAYCURTIS);
 	}
-	else if (strmatch(type, "EUCLIDEAN"))
+	else if (strmatch(type, "EUCLIDEAN") && (strmatch(dtype, "REAL") || strmatch(dtype, "SPARSEREAL")))
```
Comment (karlnapf) [inline src/shogun/ui/SGInterface.cpp]: uh these fuckers, we should really aim to get rid of those interfaces ;) LATER

File: tests/unit/distance/EuclideanDistance_unittest.cc [hunk 4]
```diff
@@ -148,3 +148,26 @@ TEST(EuclideanDistance, heterogenous_features)
 	SG_UNREF(features_lhs);
 	SG_UNREF(features_rhs);
 }
+
+TEST(EuclideanDistance, sparse_features)
+{
+	CDenseFeatures<float64_t>* features_lhs=create_lhs();
```
Comment (karlnapf) [inline tests/unit/distance/EuclideanDistance_unittest.cc]: Can you test this with a really high dimensional and really sparse set of vectors?
To make sure it doesnt blow up memory?
Like 10mio dims with 100 components nonzero

**Comment by karlnapf** [comment]
Fine to merge, should be added to the NEWS API changes

File: src/shogun/distance/SparseEuclideanDistance.cpp (hunk 1 above)
Comment (lambday) [inline src/shogun/distance/SparseEuclideanDistance.cpp]: euclidean distance class does this now also, i.e. precomputes the squared lhs/rhs norms.

File: tests/unit/distance/EuclideanDistance_unittest.cc (hunk 4 above)
Comment (lambday) [inline tests/unit/distance/EuclideanDistance_unittest.cc]: yeah I'll add the test.

**Comment by karlnapf** [comment]
Can we rebase and merge this @lambday 
shouldnt be a lot of work ...

**Comment by gf712** [comment]
@lambday @karlnapf is this still relevant?

**Comment by karlnapf** [comment]
Would be great to rebase and merge this @lambday interested? :)

**Pull request closed without merging**